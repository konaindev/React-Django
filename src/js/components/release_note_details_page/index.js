import React, { Component } from "react";
import PropTypes from "prop-types";

import Breadcrumbs from "../breadcrumbs";
import Container from "../container";
import PageChrome from "../page_chrome";
import ReleaseNoteDetails from "../release_note_details";
import { formatDate } from "../../utils/formatters";

const getBreadcrumbs = releaseNote => [
  {
    text: "Releases",
    link: "/releases"
  },
  {
    text: `${releaseNote.version} Release: ${releaseNote.title} | ${formatDate(
      releaseNote.date
    )}`
  }
];

export const ReleaseNoteDetailsPage = ({ releaseNote }) => (
  <PageChrome>
    <Container>
      <Breadcrumbs breadcrumbs={getBreadcrumbs(releaseNote)} />
      <ReleaseNoteDetails releaseNote={releaseNote} />
    </Container>
  </PageChrome>
);

ReleaseNoteDetailsPage.propTypes = {
  releaseNote: PropTypes.object.isRequired
};

export default ReleaseNoteDetailsPage;
