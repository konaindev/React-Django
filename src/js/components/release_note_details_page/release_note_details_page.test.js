import renderer from "react-test-renderer";
import ReleaseNoteDetailsPage from "./index";
import { props } from "./release_note_details_page.stories";

describe("ReleaseNoteDetailsPage", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<ReleaseNoteDetailsPage {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
