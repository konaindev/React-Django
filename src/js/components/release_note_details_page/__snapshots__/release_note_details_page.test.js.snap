// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReleaseNoteDetailsPage renders correctly 1`] = `
"<PageChrome>
  <Container>
    <Breadcrumbs breadcrumbs={{...}} />
    <ReleaseNoteDetails release_note={{...}} />
  </Container>
</PageChrome>"
`;
